node/src/chain/stall_chain_fsm.rs

```diff
@@ -57,27 +70,41 @@ impl<DB: database::DB, N: Network, VM: VMExecution> StalledChainFSM<DB, N, VM> {
             acc,
         };
 
-        sm.update_tip(tip.inner().header().clone());
+        sm.update_tip(tip.inner().header());
         sm
     }
 
-    pub(crate) fn reset(&mut self, tip: Header) {
+    /// Attempts to reset the FSM state, if tip has changed
+    pub(crate) fn try_reset(&mut self, tip: &Header) {
```

Comment:
I personally don't like the methods called "try_"
At least to me they create a lot of confusion and, furthermore, there is no feedback returned to the caller

Can we stick to the previous "reset" name and return a Result (with a proper error) so we can act accordingly (even if act is just a log entry)?